init.py

```diff
@@ -180,24 +181,89 @@ def sierpinski(t, l, r, deg, preds, g):
             add_edges(l,t,preds,g)
             add_edges(t,r,preds,g)
             add_edges(r,l,preds,g)
-            
+
     def add_edges(u,v,preds,g):
         ''' Adds edges between vertices u and v for all predicates. '''
         for p in preds:
             g += [[u,p,v]]
             g += [[v,p,u]]
-            
+
     def _idgen():
         ''' Generates integer identifiers for vertices. '''
         c = 4
         while True:
             yield c
             c += 1
-            
+
     ids = _idgen()
     graph = []
-    sierpinski(1,2,3,degree,predicates, graph)   
-    with open(os.path.join(target_dir, 'sierpinskigraph_%s.txt'%(degree)), 'w') as out_file:
+    sierpinski(1,2,3,degree,predicates, graph)
+    filename = 'sierpinskigraph_%d_%s.txt' % (degree,''.join(predicates))
+    _to_file(target_dir, filename, graph)
+
+def gen_kuijpers_random(target_dir, vertices, k, labels='A B C D'):
+    '''Generate random graphs as defined on J. Kuijpers's paper "An
+    Experimental Study of Context-Free Path Query Evaluation Methods".
+    '''
+    n = vertices
+    L = labels.split(' ')
+    assert n >= k, 'It is required that vertices >= k'
+
+    def label(L):
+        '''Randomly choose a label in `L`'''
+        return random.choice(L)
+
+    def add(D,i,j,I,var):
+        '''Add an edge to `D`.'''
+        l = label(L)
+        D += [[i,l,j]]
+        D += [[j,l+"R",i]]
```

Comment:
Is it important that we must add both the forward and backward edges?

Reply:
I added the backward edges to be in compliance to other databases I had seen in the repository.
You may remove it, but we may also just ignore (not load) them if we don't need them.

Reply:
I think, that backward edges was added because of feature of some queries. So in our case it is
unnecessary, because our grammar doesnt contains special rules.